x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameUsageIT.java

```diff
@@ -47,6 +47,7 @@ public void testUsage() throws IOException {
 
         // create a transform
         createPivotReviewsTransform("test_usage", "pivot_reviews", null);
+        startAndWaitForTransform("test_usage", "pivot_reviews");
```

Comment:
@hendrikmuhs, I will have to update the usage API with this change then. I will do that and remove this line.

Reply:
Actually, @hendrikmuhs, looking at `DataFrameFeatureSet#usage`, it calls the `stats` endpoint and until `stats` are stored in a doc, we need the task to exist. I will mark this with a TODO to signify such.

Reply:
You mean to get this?

```
        assertEquals(0, XContentMapValues.extractValue("data_frame.stats.index_failures", usageAsMap));
```

Agreed, this requires the state and I think it's fine to call `startAndWaitForTransform` in order to get it.

However this test:
```
assertEquals(1, XContentMapValues.extractValue("data_frame.transforms._all", usageAsMap));
```

should work.

What about:

```
        createPivotReviewsTransform("test_usage", "pivot_reviews", null);
        
        usageResponse = client().performRequest(new Request("GET", "_xpack/usage"));

        usageAsMap = entityAsMap(usageResponse);
        // we should see the job
        assertEquals(1, XContentMapValues.extractValue("data_frame.transforms._all", usageAsMap));
       // but no stats
        assertEquals(null, XContentMapValues.extractValue("data_frame.stats", usageAsMap));

        // start it
        startAndWaitForTransform("test_usage", "pivot_reviews");
        // get usage again
        usageResponse = client().performRequest(new Request("GET", "_xpack/usage"));

        usageAsMap = entityAsMap(usageResponse);
        // we should see some stats
        assertEquals(1, XContentMapValues.extractValue("data_frame.transforms._all", usageAsMap));
        assertEquals(0, XContentMapValues.extractValue("data_frame.stats.index_failures", usageAsMap));
        assertEquals(???, XContentMapValues.extractValue("data_frame.stats.documents_indexed", usageAsMap));
```

Reply:
Sounds good. I will add a case for when there are no running transforms.

Reply:
@hendrikmuhs , what makes this difficult is that there is no way to get stats at all outside of the allocated persistent task as it is stored in `Indexer.getStats()`. Consequently, if the process crashed and had to move to another node, all previous stats would be lost.

I will refactor this the usage endpoint, but it will immediately have to be refactored again when we store stats in something outside of the running state of the allocated task on the node.

Reply:
Additionally, I am not sure you want to get all non-running transforms...a hearkening back to your worries of 10k+ configurations.

Reply:
:+1: and now the fun begins! Excellent find! This is called regularly if usage collection is enabled. 

If I remember correctly this summarizes the overall transforms and transforms per state. Makes sense to me, I think we should keep doing this. But now that we store everything in an index we can re-factor (this is pre-index code).

Anyway, can you put this into an issue? No need to solve this as part of this PR. 

Additionally I think this isn't only a problem here, other parts of the code count in a similar fashion.

Reply:
@hendrikmuhs definitely, I will open an issue. I think if we move stats to an index, aggregations will solve this for us, there may be some wonkiness around transforms that are yet to have a task created (do we create a blank stats document when the transform is created?).